core/src/main/java/org/bitcoinj/store/DatabaseFullPrunedBlockStore.java

```diff
@@ -619,7 +619,9 @@ protected void putUpdateStoredBlock(StoredBlock storedBlock, boolean wasUndoable
             s.setBytes(1, hashBytes);
             s.setBytes(2, storedBlock.getChainWork().toByteArray());
             s.setInt(3, storedBlock.getHeight());
-            s.setBytes(4, storedBlock.getHeader().unsafeBitcoinSerialize());
+            s.setBytes(4, storedBlock.getHeader().getTransactions() == null ?
```

Comment:
Bloody hell agreed  (Sry that was a lesson in how to muck up every line you change - however after this weekend it is now so much clearer).
I’ve changed and pushed.
K

> On 23 Jun 2015, at 20:40, Mike Hearn <notifications@github.com <mailto:notifications@github.com>> wrote:
> 
> In core/src/main/java/org/bitcoinj/store/DatabaseFullPrunedBlockStore.java <https://github.com/bitcoinj/bitcoinj/pull/994#discussion_r33082519>:
> 
> > @@ -619,7 +619,9 @@ protected void putUpdateStoredBlock(StoredBlock storedBlock, boolean wasUndoable
> >              s.setBytes(1, hashBytes);
> >              s.setBytes(2, storedBlock.getChainWork().toByteArray());
> >              s.setInt(3, storedBlock.getHeight());
> > -            s.setBytes(4, storedBlock.getHeader().unsafeBitcoinSerialize());
> > +            s.setBytes(4, storedBlock.getHeader().getTransactions() == null ?
> Ditto. This is a full store: getTransactions() should never be null, I think? So it needs to always be run through cloneAsHeader()
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/bitcoinj/bitcoinj/pull/994/files#r33082519>.
> 




Kalpesh Parmar
Tel:  +44 7474575321 (UK)